homeassistant/components/weatherflow/sensor.py

```diff
@@ -0,0 +1,378 @@
+        suggested_display_precision=2,
+        raw_data_conv_fn=lambda raw_data: raw_data.magnitude,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="wind_speed_average",
+        translation_key="wind_speed_average",
+        icon="mdi:weather-windy",
+        device_class=SensorDeviceClass.WIND_SPEED,
+        native_unit_of_measurement=UnitOfSpeed.KILOMETERS_PER_HOUR,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=2,
+        raw_data_conv_fn=lambda raw_data: raw_data.magnitude,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="wind_direction",
+        translation_key="wind_direction",
+        icon="mdi:compass-outline",
+        native_unit_of_measurement=DEGREE,
+        state_class=SensorStateClass.MEASUREMENT,
+        event_subscriptions=[EVENT_RAPID_WIND, EVENT_OBSERVATION],
+        raw_data_conv_fn=lambda raw_data: raw_data.magnitude,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="wind_direction_average",
+        translation_key="wind_direction_average",
+        icon="mdi:compass-outline",
+        native_unit_of_measurement=DEGREE,
+        state_class=SensorStateClass.MEASUREMENT,
+        raw_data_conv_fn=lambda raw_data: raw_data.magnitude,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up WeatherFlow sensors using config entry."""
+
+    @callback
+    def async_add_sensor(device: WeatherFlowDevice) -> None:
+        """Add WeatherFlow sensor."""
+        LOGGER.debug("Adding sensors for %s", device)
+
+        sensors: list[WeatherFlowSensorEntity] = [
+            WeatherFlowSensorEntity(
+                device=device,
+                description=description,
+                is_metric=(hass.config.units == METRIC_SYSTEM),
+            )
+            for description in SENSORS
+            if hasattr(device, description.key)
+        ]
+
+        async_add_entities(sensors)
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(
+            hass,
+            format_dispatch_call(config_entry),
+            async_add_sensor,
+        )
+    )
+
+
+class WeatherFlowSensorEntity(SensorEntity):
+    """Defines a WeatherFlow sensor entity."""
+
+    entity_description: WeatherFlowSensorEntityDescription
+    _attr_should_poll = False
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        device: WeatherFlowDevice,
+        description: WeatherFlowSensorEntityDescription,
+        is_metric: bool = True,
+    ) -> None:
+        """Initialize a WeatherFlow sensor entity."""
+        self.device = device
+        self.entity_description = description
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, device.serial_number)},
+            manufacturer="WeatherFlow",
+            model=device.model,
+            name=f"{device.serial_number}",
```

Comment:
```suggestion
            name=str(device.serial_number),
```
Unless it's already a string